### PR TITLE
Add the ability to do JSON escaping of printed file and function names.

### PR DESCRIPTION
**Pull request by nnethercote**
This is necessary for DMD, which puts file names and function names
within JSON strings.

For testing I had to add a new test binary with the awkward name of
`example"json`, which requires escaping of the '"' char.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -225,23 +240,29 @@ impl Fixer {
             },
         };
 
-        if let Some(func_info) = file_info.func_info(address) {
-            let name = func_info.demangled_name();
-            if let Some(line_info) = func_info.line_info(address) {
+        let (mut fn_name, mut location) = if let Some(func_info) = file_info.func_info(address) {
```
Comment (gabrielesvelto) [inline src/main.rs]: Double assignments on one line can be confusing, please split the logic that finds the function name and location to a separate function.

**Review by gabrielesvelto — CHANGES_REQUESTED** [review]
Looking good save for one small change.

File: src/main.rs (hunk 1 above)
Comment (nnethercote) [reply src/main.rs]: I have changed it a bit so the double assignment isn't necessary.

**Pull request merged**